src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -92,31 +127,10 @@ private void ActivateProperties(object controller, Type containerType, Dictionar
             modelState.MaxAllowedErrors = _options.MaxModelValidationErrors;
             foreach (var parameter in parameterMetadata)
             {
-                var metadata = _modelMetadataProvider.GetMetadataForType(parameter.ParameterType);
-                var parameterType = parameter.ParameterType;
-                var modelBindingContext = GetModelBindingContext(
-                    parameter.Name,
-                    metadata,
-                    parameter.BindingInfo,
-                    modelState,
-                    operationContext);
-
-                var modelBindingResult = await operationContext.ModelBinder.BindModelAsync(modelBindingContext);
-                if (modelBindingResult != null && modelBindingResult.IsModelSet)
+                var model = await BindModelAsync(parameter, modelState, operationContext);
+                if (model != null)
```

Comment:
Note that this is a change in behavior. What happens if the model was bound, but the value was null - now there's no entry in the dictionary. I think we need to return the MBR instead.